Cannot run, crashes

Up-to-date Win 10 machine. Just installed Python3.8.2, pip, setuptools, winpcap. Followed guide.  Setup.py install runs fine then this:

```
(env) C:\Boofuzz>python network_monitor.py
Traceback (most recent call last):
  File "network_monitor.py", line 13, in <module>
    import pcapy  # pytype: disable=import-error
ModuleNotFoundError: No module named 'pcapy'
```

Try to install pcapy through pip:
```
(env) C:\Boofuzz>pip3 install pcapy
Collecting pcapy
  Using cached pcapy-0.11.4.tar.gz (37 kB)
Installing collected packages: pcapy
    Running setup.py install for pcapy ... error
    ERROR: Command errored out with exit status 1:
     command: 'c:\env\scripts\python.exe' -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\myuser\\AppData\\Local\\Temp\\pip-install-d9u__v51\\pcapy\\setup.py'"'"'; __file__='"'"'C:\\Users\\myuser\\AppData\\Local\\Temp\\pip-install-d9u__v51\\pcapy\\setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record 'C:\Users\myuser\AppData\Local\Temp\pip-record-33d9so_5\install-record.txt' --single-version-externally-managed --compile --install-headers 'c:\env\include\site\python3.8\pcapy'
         cwd: C:\Users\myuser\AppData\Local\Temp\pip-install-d9u__v51\pcapy\
    Complete output (12 lines):
    running install
    running build
    running build_ext
    building 'pcapy' extension
    creating build
    creating build\temp.win-amd64-3.8
    creating build\temp.win-amd64-3.8\Release
    creating build\temp.win-amd64-3.8\Release\win32
    C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.22.27905\bin\HostX86\x64\cl.exe /c /nologo /Ox /W3 /GL /DNDEBUG /MD -DWIN32=1 -Ic:\wpdpack\Include -Ic:\env\include -IC:\Python\Python38\include -IC:\Python\Python38\include "-IC:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.22.27905\ATLMFC\include" "-IC:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.22.27905\include" "-IC:\Program Files (x86)\Windows Kits\NETFXSDK\4.7.2\include\um" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\ucrt" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\shared" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\um" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\winrt" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\cppwinrt" /EHsc /Tppcapdumper.cc /Fobuild\temp.win-amd64-3.8\Release\pcapdumper.obj
    pcapdumper.cc
    pcapdumper.cc(11): fatal error C1083: Cannot open include file: 'pcap.h': No such file or directory
    error: command 'C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\Community\\VC\\Tools\\MSVC\\14.22.27905\\bin\\HostX86\\x64\\cl.exe' failed with exit status 2
    ----------------------------------------
ERROR: Command errored out with exit status 1: 'c:\env\scripts\python.exe' -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\myuser\\AppData\\Local\\Temp\\pip-install-d9u__v51\\pcapy\\setup.py'"'"'; __file__='"'"'C:\\Users\\myuser\\AppData\\Local\\Temp\\pip-install-d9u__v51\\pcapy\\setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record 'C:\Users\myuser\AppData\Local\Temp\pip-record-33d9so_5\install-record.txt' --single-version-externally-managed --compile --install-headers 'c:\env\include\site\python3.8\pcapy' Check the logs for full command output.
```

Try to install pcapy from source:

```
(env) C:\pcapy-0.11.5>python setup.py install
C:\env\lib\site-packages\setuptools\dist.py:471: UserWarning: Normalizing '0.11.5-dev' to '0.11.5.dev0'
  warnings.warn(
running install
running bdist_egg
running egg_info
writing pcapy.egg-info\PKG-INFO
writing dependency_links to pcapy.egg-info\dependency_links.txt
writing top-level names to pcapy.egg-info\top_level.txt
C:\pcapy-0.11.5\WpdPack_4_1_2\WpdPack\Include\bittypes.h(124): note: see previous definition of 'PRIx64'
pcapy.cc(262): error C2375: 'PyInit_pcapy': redefinition; different linkage
C:\pcapy-0.11.5\pcapy.h(23): note: see declaration of 'PyInit_pcapy'
error: command 'C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\Community\\VC\\Tools\\MSVC\\14.22.27905\\bin\\HostX86\\x64\\cl.exe' failed with exit status 2
```

I've tried running from cmd, VS2008 cmd, VS2017 cmd, VS2019 cmd, all 32 and 64 bit versions.

--- a/CHANGELOG.rst
+++ b/CHANGELOG.rst
@@ -26,6 +26,7 @@ Fixes
 - Refactored old unit-tests
 - Removed deprecated session arguments `fuzz_data_logger`, `log_level`, `logfile`, `logfile_level` and `log()`.
 - Removed deprecated logger `FuzzLoggerFile`.
+- Fixed network monitor compatibility with Python 3.
 
 v0.1.6
 ------

--- a/INSTALL.rst
+++ b/INSTALL.rst
@@ -177,6 +177,12 @@ The network monitor was Sulley’s primary tool for recording test data,
 and has been replaced with boofuzz’s logging mechanisms.
 However, some people still prefer the PCAP approach.
 
+.. note::
+    The network monitor requires Pcapy, which will not be automatically installed with boofuzz. You can manually
+    install it with ``pip install pcapy``.
+
+    If you run into errors, check out the requirements on the `project page <https://github.com/helpsystems/pcapy>`_.
+
 .. _help site: http://www.howtogeek.com/197947/how-to-install-python-on-windows/
 .. _releases page: https://github.com/jtpereyda/boofuzz/releases
 .. _`https://github.com/jtpereyda/boofuzz`: https://github.com/jtpereyda/boofuzz

--- a/network_monitor.py
+++ b/network_monitor.py
@@ -60,19 +60,19 @@ Network Device List:
     for index, pcapy_device in enumerate(ifs):
         # if we are on windows, try and resolve the device UUID into an IP address.
         if sys.platform.startswith("win"):
-            import _winreg  # pytype: disable=import-error
+            from six.moves import winreg  # pytype: disable=import-error
 
             try:
                 # extract the device UUID and open the TCP/IP parameters key for it.
                 pcapy_device = pcapy_device[pcapy_device.index("{") : pcapy_device.index("}") + 1]
                 subkey = r"SYSTEM\CurrentControlSet\Services\Tcpip\Parameters\Interfaces\%s" % pcapy_device
-                key = _winreg.OpenKey(_winreg.HKEY_LOCAL_MACHINE, subkey)
+                key = winreg.OpenKey(winreg.HKEY_LOCAL_MACHINE, subkey)
 
                 # if there is a DHCP address snag that, otherwise fall back to the IP address.
                 try:
-                    ip = _winreg.QueryValueEx(key, "DhcpIPAddress")[0]
+                    ip = winreg.QueryValueEx(key, "DhcpIPAddress")[0]
                 except Exception:
-                    ip = _winreg.QueryValueEx(key, "IPAddress")[0][0]
+                    ip = winreg.QueryValueEx(key, "IPAddress")[0][0]
 
                 pcapy_device = pcapy_device + "\t" + ip
             except Exception:

